pkgs/racket-doc/scribblings/reference/sequences.scrbl

```diff
@@ -746,58 +746,67 @@ each element in the sequence.
                                    (any/c . -> . any/c)
                                    any/c
                                    (or/c (any/c . -> . any/c) #f)
-                                   (or/c (() () #:rest list? . ->* . any/c) #f)
-                                   (or/c ((any/c) () #:rest list? . ->* . any/c) #f)))
+                                   (or/c (any/c ... . -> . any/c) #f)
+                                   (or/c (any/c any/c ... . -> . any/c) #f)))
                        (-> (values (any/c . -> . any)
                                    (or/c (any/c . -> . any/c) #f)
                                    (any/c . -> . any/c)
                                    any/c
                                    (or/c (any/c . -> . any/c) #f)
-                                   (or/c (() () #:rest list? . ->* . any/c) #f)
-                                   (or/c ((any/c) () #:rest list? . ->* . any/c) #f))))])
+                                   (or/c (any/c ... . -> . any/c) #f)
+                                   (or/c (any/c any/c ... . -> . any/c) #f))))])
          sequence?]{
-  Returns a sequence whose elements are generated by the procedures
-  and initial value returned by the thunk, which is called to
-  @tech{initiate} the sequence.  The initiated sequence is defined in
-  terms of a @defterm{position}, which is initialized to the third
-  result of the thunk, and the @defterm{element}, which may consist of
-  multiple values.
-
-  The @racket[thunk] results define the generated elements as follows:
+  Returns a sequence whose elements are generated according to @racket[thunk].
+
+  The sequence is @tech{initiate}d when @racket[thunk] is called.
+  The initiated sequence is defined in
+  terms of a @defterm{position}, which is initialized to @racket[_init-pos],
+  and the @defterm{element}, which may consist of multiple values.
+
+  @racket[thunk] must return either 6 or 7 values.
```

Comment:
To avoid starting the sentence with a lowercased variable, I'd write "The @racket[thunk] procedure" here. Also, this document generally follows the rule of spelling out numbers less than 10 in prose.